### PR TITLE
bfd-topo1: add basic topology test for `bfdd`

### PR DESCRIPTION
**Pull request by rzalamena**
This PR implements a topology test for the new `bfdd` daemon ( https://github.com/FRRouting/frr/pull/2294 ).

The following tests are done:
* Validate that BFD peers connect and properly detect neighbors configurations;
* BGP connects and redistribute prefixes;
* BFD status changes when a link fails;
* BGP converges quickly after the link failure;

### CODE REVIEW

**Comment by louberger** [comment]
@rzalamena can you rebase when you have the chance

File: bfd-topo1/test_bfd_topo1.py [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+    tgen.stop_topology()
+
+
+def test_bfd_connection():
+    "Assert that the BFD peers can find themselves."
+    tgen = get_topogen()
+    if tgen.routers_have_failure():
+        pytest.skip(tgen.errors)
+
+    topotest.sleep(2, 'waiting for bfd peers to go up')
```
[the remark this answers is not in the document]
Comment (rzalamena) [reply bfd-topo1/test_bfd_topo1.py]: This should not be a problem the BFD protocol converges in less than a second, that was added as a safety mesure but probably not needed. If it takes more than 3 seconds then `bfdd` already failed.

It seems that recording the time taken for a function is a repeating pattern that we should think about implementing in topotests/topogen.

**Comment by rzalamena** [comment]
> @louberger 
> rebase is needed. Would prefer sleeps removed as mentioned in specific comments - but this is a 'non-blocking;' comment.

I rebased the branch, but I'll fix the sleep as next thing. I'm thinking about something to record the time taken to converge and I'll come back later.

Thanks for the review!

**Comment by donaldsharp** [comment]
@louberger is this fixed to your satisfaction now?

**Comment by rzalamena** [comment]
Rebased with `master`. FRR also has `bfdd` in `master` as well.

**Pull request merged**